main.js

```diff
@@ -134,246 +78,271 @@ var getBlocks = () => {
+                        return true;
+                    }
+                }
+            })
+        }
+
+        sortedForgedBlocks[0].voterBalances = new Map(curBals.map(vote => [vote.address, { // Populate the highest block
+            'balance': vote.balance,
+            'share': 0
+        }]));
+        txs.forEach((tx, idx) => {
+            sortTx(tx);
+        });
+        // Populate the last blocks we missed
+        sortedForgedBlocks.forEach((block, idx) => {
+            if (!block.voterBalances) {
+                block.voterBalances = new Map(sortedForgedBlocks[idx - 1].voterBalances);
+            }
+        })
+        currentVoters = new Set(voterAddrs);
+        let voterSum = 0;
+        latestForgedBlock.voterBalances.forEach((bal) => voterSum += bal[0])
+        console.log("Pool balance calculations complete");
+
+        resolve();
+    })
+};
+
+let getVoterWeight = () => {
+    console.log("Calculating voter weights for each block...");
+    let taxes = new BigNumber(0);
+    let totalPay = new BigNumber(0);
+    return new Promise((resolve, rej) => {
+        sortedForgedBlocks = sortedForgedBlocks.reverse();
+        sortedForgedBlocks.forEach((block, idx) => {
+            //console.log("block - " + parseInt(idx + 1) + " / " + sortedForgedBlocks.length + " | blockVoterBal.size: " + block.voterBalances.size);
+
+            block.voterBalances.forEach((balanceData, index) => {
+                if (balanceData.balance > (config.cap * 100000000)) {
+                    let curData = sortedForgedBlocks[idx].voterBalances.get(index);
+                    curData.balance = (config.cap * 100000000);
+                    curData.overlimit = true;
+                    sortedForgedBlocks[idx].voterBalances.set(index, curData);
+                }
             });
 
-            //compute voter shares
+            const poolTotal = [...block.voterBalances.values()].map(val => val.balance).reduce((a, b) => a + b);
+
             block.voterBalances.forEach((balanceData, addr) => {
-                if (balanceData[0] > 0) {
-                    //[paidShare, tax]
-                    var share = blockShareFunc(poolTotal, balanceData[0]);
-                    balanceData[1] = share[0];
+
+                //let curData = sortedForgedBlocks[idx].voterBalances.get(addr);
+                let max = new BigNumber(balanceData.balance);
+
+                if (!fullBalances.includes(addr)) {
+                    max = max.times(config.payout);
+                }
+
+                balanceData.max = max.toNumber();
+
+                if (balanceData.balance > 1) {
+                    let share = utils.blockShareFunc(poolTotal, balanceData.max);
+                    balanceData.share = share[0];
 
                     //If they are blacklisted, keep their share
-                    if (blacklist[addr]) {
-                        balanceData[1] = new BigNumber(0);
+                    if (blacklist.includes(addr)) {
```

Comment:
You turned blacklist from an object to an array which is not backwards compatible. If you're going to do this, update the documentation.